fenix/app/src/main/java/org/mozilla/fenix/compose/ThumbnailCard.kt

```diff
@@ -62,72 +57,35 @@ fun ThumbnailCard(
         modifier = modifier,
         backgroundColor = backgroundColor,
     ) {
-        if (inComposePreview) {
-            Box(
-                modifier = Modifier.background(color = FirefoxTheme.colors.layer3),
-            )
-        } else {
-            components.core.icons.Loader(url) {
-                Placeholder {
-                    Box(
-                        modifier = Modifier.background(color = FirefoxTheme.colors.layer3),
-                    )
-                }
-
-                WithIcon { icon ->
-                    Box(
-                        modifier = Modifier.size(36.dp),
-                        contentAlignment = Alignment.Center,
-                    ) {
-                        Image(
-                            painter = icon.painter,
-                            contentDescription = contentDescription,
-                            modifier = Modifier
-                                .size(36.dp)
-                                .clip(RoundedCornerShape(8.dp)),
-                            contentScale = contentScale,
-                        )
-                    }
-                }
-            }
-
-            ThumbnailImage(
-                key = key,
-                size = size,
-                modifier = modifier,
-                contentScale = contentScale,
-                alignment = alignment,
-            )
-        }
-    }
-}
-
-@Composable
-private fun ThumbnailImage(
-    key: String,
-    size: Dp,
-    modifier: Modifier,
-    contentScale: ContentScale,
-    alignment: Alignment,
-) {
-    val rememberBitmap = remember(key) { mutableStateOf<ImageBitmap?>(null) }
-    val thumbnailSize = LocalDensity.current.run { size.toPx().toInt() }
-    val request = ImageLoadRequest(key, thumbnailSize)
-    val storage = components.core.thumbnailStorage
-    val bitmap = rememberBitmap.value
-
-    LaunchedEffect(key) {
-        rememberBitmap.value = storage.loadThumbnail(request).await()?.asImageBitmap()
-    }
-
-    if (bitmap != null) {
-        val painter = BitmapPainter(bitmap)
-        Image(
-            painter = painter,
-            contentDescription = null,
+        ThumbnailImage(
+            key = key,
+            size = size,
             modifier = modifier,
             contentScale = contentScale,
             alignment = alignment,
+            fallbackContent = {
```

Comment:
optional nit: I almost feel like we could move this argument out of the list here and treat it the same way the that, for example, the `content` parameter for `Box`es are usually treated.